File: Complementar - Estruturas de dados (Didático)/Algoritmos09/Alg09.php
<?php
/*
♦ Algoritmo 09: Estrutura de Dados Linear Estática (não Dinâmica) de Pesquisa/Busca Pesquisa Sequencial (sem a utilização de variável/sentinela).
*/

function buscaSequencial($array, $valorProcurado) {
    for ($i = 0; $i < count($array); $i++) {
        if ($array[$i] === $valorProcurado) {
            return $i; // Retorna o índice do valor procurado
        }
    }
    return -1; // Retorna -1 se o valor não foi encontrado
}

$array = [10, 5, 2, 8, 3, 7, 1, 6, 9, 4];
$valorProcurado = 7;

$indiceEncontrado = buscaSequencial($array, $valorProcurado);

if ($indiceEncontrado !== -1) {
    echo "Valor " . $valorProcurado . " encontrado no índice: " . $indiceEncontrado;
} else {
    echo "Valor " . $valorProcurado . " não encontrado no array.";
}

?>

File: Complementar - Estruturas de dados (Didático)/Algoritmos09/Alg09.java
/*
♦ Algoritmo 09: Algoritmo de Pesquisa/Busca para Estruturas de Dados Lineares Estáticas (não Dinâmicas): Busca Sequencial (sem a utilização de variável/sentinela).
*/

public class Alg09 {
    
    public static int buscaSequencial(int[] array, int valorProcurado) {
        for (int i = 0; i < array.length; i++) {
            if (array[i] == valorProcurado) {
                return i; // Retorna o índice do valor procurado
            }
        }
        return -1; // Retorna -1 se o valor não foi encontrado
    }

    public static void main(String[] args) {
        int[] array = {10, 5, 2, 8, 3, 7, 1, 6, 9, 4};
        int valorProcurado = 7;
        
        int indiceEncontrado = buscaSequencial(array, valorProcurado);
        
        if (indiceEncontrado != -1) {
            System.out.println("Valor " + valorProcurado + " encontrado no índice: " + indiceEncontrado);
        } else {
            System.out.println("Valor " + valorProcurado + " não encontrado no array.");
        }
    }
}
